charts/multicluster-controlplane/templates/secret.yaml

```diff
@@ -9,30 +9,30 @@ stringData:
     apiserver:
       externalHostname: {{ .Values.apiserver.externalHostname }}
       port: 443
-      {{- if and .Values.apiserver.caFile .Values.apiserver.caKeyFile }}
-      caFile: "/controlplane_config/{{ .Values.apiserver.caFile | base }}"
-      caKeyFile: "/controlplane_config/{{ .Values.apiserver.caKeyFile | base }}"
+      {{- if (and .Values.apiserver.ca .Values.apiserver.cakey) }}
+      caFile: "/controlplane_config/apiserver_ca.crt"
+      caKeyFile: "/controlplane_config/apiserver_ca.key"
       {{- end }}
     etcd:
       mode: {{ .Values.etcd.mode }}
       prefix: {{ .Release.Namespace }}
-      {{- if and (eq .Values.etcd.mode "external") (and .Values.etcd.caFile (and .Values.etcd.certFile (and .Values.etcd.keyFile (not (empty .Values.etcd.servers))))) }}
+      {{- if (and (eq .Values.etcd.mode "external") (and (and .Values.etcd.ca  (and .Values.etcd.cert .Values.etcd.certkey)) (not (empty .Values.etcd.servers)))) }}
       servers:
-      {{- range $index, $value := .Values.etcd.servers }}
-      - {{ $value }}
+      {{- range .Values.etcd.servers }}
+      - {{ . }}
       {{- end }}
-      caFile: "/controlplane_config/{{ .Values.etcd.caFile }}"
-      certFile: "/controlplane_config/{{ .Values.etcd.certFile }}"
-      keyFile: "/controlplane_config/{{ .Values.etcd.keyFile }}"
+      caFile: "/controlplane_config/etcd_ca.crt"
+      certFile: "/controlplane_config/etcd_cert.crt"
+      keyFile: "/controlplane_config/etcd_cert.key"
       {{- end }}
 
-data:
-{{- if and .Values.apiserver.caFile .Values.apiserver.caKeyFile }}
-{{ (.Files.Glob .Values.apiserver.caFile).AsSecrets | indent 2  }}
-{{ (.Files.Glob .Values.apiserver.caKeyFile).AsSecrets | indent 2 }}
-{{- end }}
-{{- if and (eq .Values.etcd.mode "external") ( and (and .Values.etcd.caFile (and .Values.etcd.certFile .Values.etcd.keyFile)) (not (empty .Values.etcd.servers)) ) }}
-{{ (.Files.Glob .Values.etcd.caFile).AsSecrets | indent 2 }} 
-{{ (.Files.Glob .Values.etcd.certFile).AsSecrets | indent 2 }}
-{{ (.Files.Glob .Values.etcd.keyFile).AsSecrets | indent 2 }}
-{{- end}}
+  {{- if (and .Values.apiserver.ca .Values.apiserver.cakey) }}
+  apiserver_ca.crt: {{ .Values.apiserver.ca | quote  }}
+  apiserver_ca.key: {{ .Values.apiserver.cakey | quote  }}
+  {{- end }}
+  {{- if (and (eq .Values.etcd.mode "external") (and (and .Values.etcd.ca (and .Values.etcd.cert .Values.etcd.certkey)) (not (empty .Values.etcd.servers)))) }}
```

Comment:
to keep simple, can we just use `{{- if (eq .Values.etcd.mode "external")}}` here?